parsec/core/fs/workspacefs/workspacefile.py

```diff
@@ -0,0 +1,229 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
+
+import os
+import re
+from enum import Enum
+import functools
+from parsec.core.fs.exceptions import FSUnsupportedOperation, FSOffsetError
+from parsec.core.types import FsPath
+
+
+class FileState(Enum):
+    INIT = 0
+    OPEN = 1
+    CLOSED = 2
+
+
+def check_state(func):
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        if self.closed:
+            raise ValueError("I/O operation on closed file.")
+
+        return func(self, *args, **kwargs)
+
+    return wrapper
+
+
+class WorkspaceFile:
+    """Async file Object.
+
+    Methods of this object should act like trio async file :
+    "https://github.com/python-trio/trio/blob/master/trio/_file_io.py" using transactions.
+
+    Keyword arguments:
+    fd -- the file descriptor.
+    transactions -- object to use for file transactions.
+    path -- relative path of the file.
+    mode -- the mode where the file have been opened.
+    mode("a") -> append file.
+    mode("b") -> bytes, have to be used with another mode, ignored, class is already working with
+    bytes.
+    mode("w") -> write file.
+    mode("r") -> read file.
+    """
+
+    def __init__(self, fd: int, transactions, path: FsPath, mode: str = "r"):
+        self._fd = fd
+        self._name = None
```

Comment:
Unused?

Reply:
It was suppose to be implemented later but finaly the `name` method return the path. I remove this attribute